### PR TITLE
[CM-829] Add Typing Indicator for Welcome Message | iOS

### PR DESCRIPTION
**Pull request by Sathyan-Elangovan**
## Summary
- Added Typing Indicator for Welcome Message

### CODE REVIEW

File: Sources/ViewModels/ALKConversationViewModel.swift [hunk 1]
```diff
@@ -1203,10 +1203,16 @@ open class ALKConversationViewModel: NSObject, Localizable {
...
             NSLog("messages loaded: ", messages)
             self.alMessages = messages.reversed() as! [ALMessage]
-            self.alMessageWrapper.addObject(toMessageArray: messages)
-            let models = self.alMessages.map { $0.messageModel }
-            self.messageModels = models
+            let contactService = ALContactService()
 
+            if let alContact = contactService.loadContact(byKey: "userId", value:  self.alMessages[0].to),
+              let role = alContact.roleType,
+              role !=  NSNumber.init(value: AL_BOT.rawValue) {
```
Comment (mukeshthawani) [inline Sources/ViewModels/ALKConversationViewModel.swift:1210]: We already have a function to get the role, use that. We can create one if it's not present.

**Review by mukeshthawani — COMMENTED** [review]
Build is failing

File: Sources/Controllers/ALKConversationViewController.swift [hunk 2]
```diff
@@ -375,7 +374,7 @@ open class ALKConversationViewController: ALKBaseViewController, Localizable {
     }
     
     open func addMessagesToList(_ messageList: [Any]) {
-        viewModel.addMessagesToList(messageList)
+       // viewModel.addMessagesToList(messageList)
```
Comment (mukeshthawani) [inline Sources/Controllers/ALKConversationViewController.swift:377]: Remove commented code

File: Sources/Controllers/ALKConversationViewController.swift [hunk 3]
```diff
@@ -1013,7 +1005,7 @@ open class ALKConversationViewController: ALKBaseViewController, Localizable {
         UIMenuController.shared.setMenuVisible(false, animated: true)
         hideMoreBar()
     }
-
+
```
Comment (mukeshthawani) [inline Sources/Controllers/ALKConversationViewController.swift:1008]: Whitespace

File: Sources/ViewModels/ALKConversationViewModel.swift [hunk 4]
```diff
@@ -1184,7 +1192,10 @@ open class ALKConversationViewModel: NSObject, Localizable {
         conversationProfile.status = ALKConversationProfile.Status(isOnline: contact.connected, lastSeenAt: contact.lastSeenAt)
         return conversationProfile
     }
-
+    let contactService = ALContactService()
+    var timer = Timer()
+    var count = 0
```
Comment (mukeshthawani) [inline Sources/ViewModels/ALKConversationViewModel.swift:1197]: Avoid global vars. If it's really needed then change the name.

File: Sources/ViewModels/ALKConversationViewModel.swift [hunk 5]
```diff
@@ -1215,6 +1232,37 @@ open class ALKConversationViewModel: NSObject, Localizable {
             }
         })
     }
+    
+    func loopOverTheLoadedMessageArray() {
+        if count >= alMessages.count {
+            return
+        }
```
Comment (mukeshthawani) [inline Sources/ViewModels/ALKConversationViewModel.swift:1239]: It's not clear what the function is doing.

Comment (Sathyan-Elangovan) [reply Sources/ViewModels/ALKConversationViewModel.swift:1239]: this method is to show typing indicator for Welcome message.

Comment (mukeshthawani) [reply Sources/ViewModels/ALKConversationViewModel.swift:1245]: Then change its name to `showTypingIndicatorForWelcomeMessage` or something. Or make it generic and use it for both welcome message and normal message.

File: Sources/ViewModels/ALKConversationViewModel.swift [hunk 6]
```diff
@@ -64,6 +64,9 @@ open class ALKConversationViewModel: NSObject, Localizable {
     // MARK: - Outputs
 
     open var isFirstTime = true
+    var timer = Timer()
+    var messagePosition = 0
```
Comment (mukeshthawani) [inline Sources/ViewModels/ALKConversationViewModel.swift:68]: In this ViewModel we've a list of messages, so which message's position is it? Can you rename it based on that?

Comment (Sathyan-Elangovan) [reply Sources/ViewModels/ALKConversationViewModel.swift]: @mukeshthawani  updated

File: Sources/ViewModels/ALKConversationViewModel.swift (hunk 5 above)
Comment (Sathyan-Elangovan) [reply Sources/ViewModels/ALKConversationViewModel.swift:1245]: updated

**Review by mukeshthawani — APPROVED** [review]

**Pull request merged**